ERCS/erc-7831.md

```diff
@@ -0,0 +1,367 @@
+
+These components may be enclosed by parentheses (`(` and `)`) to resolve parsing ambiguities.
+
+More formally, valid chain-specific addresses MUST adhere to the following grammar:
+
+```abnf
+address         = OPEN bare-address CLOSE /
+                  bare-address
+
+bare-address    = local-part SEP chain-part [SEP root-part]
+
+OPEN            = '('
+CLOSE           = ')'
+SEP             = ':'
+```
+
+#### Local Part
+
+The `local-part` is the most specific section of a chain-specific address. It identifies the account on the target chain. It can be either a hexadecimal string (`hex-address`) or an ENS-like name (`ens-like`).
+
+Valid `local-part` fragments MUST match the following grammar:
+
+```abnf
+local-part      = hex-address /
+                  ens-like
+```
+
+##### Hexadecimal Address
+
+When `local-part` is a hexadecimal string, it MUST include checksum letter casing (see [Checksum](#checksum)), and it MAY omit leading zeros. It MUST NOT include a leading `0x` prefix.
+
+Note that `local-part` may encode an address longer or shorter than 20 bytes (40 hexadecimal digits.) Implementations MUST support `local-part` lengths of 1 hexadecimal digit up to 40 digits. Implementations SHOULD support arbitrarily sized (within some reasonable limit) `local-part` components.
+
+Formally, `hex-address` MUST match the following grammar:
+
+```abnf
+hex-address     = 1*HEXDIG
+```
+
+##### ENS-Like Names
+
+To disambiguate an ENS name from a hexadecimal string—and unlike standard ENS names—names used in the `local-part` of a chain-specific address MUST contain at least one dot (`.`). If present, a dot placed at the rightmost position (eg. `eth.` or `example.eth.`) SHALL be removed before resolving the name. Chain-specific addresses SHOULD NOT contain a dot in the rightmost position unless no other dot is present.
+
+The following grammar is illustrative only. See [ERC-137] for the definition of an ENS name.
+
+```abnf
+; Rough approximation of ENS names, with the additional requirement that it
+; contain at least one "."
+ens-like        = 1*NOTSEP DOT *(1*NOTSEP [DOT])
+
+NOTSEP          = %x01-39 / %x3b-ff
+```
+
+#### Chain Part
+
+The `chain-part` identifies the target chain. It MUST be a valid ENS name as defined in [ERC-137].
+
+The following grammar is illustrative only. See [ERC-137] for the definition of an ENS name.
+
+```abnf
+chain-part      = ens-name
+
+; Rough approximation of ENS names, with no additional requirements
+ens-name        = 1*NOTSEP
+```
+
+#### Root Part
+
+The `root-part` identifies the root chain against which other names are resolved. When present, the `root-part` SHALL be the [EIP-155] `chainid` of the root chain in decimal format. `root-part` SHOULD NOT be present when `chainid == 1`, and MUST be present when `chainid != 1`.
+
+```abnf
+root-part = 1*DIGIT
+```
+
+### Resolution
+
+Resolving a chain-specific address begins on the right, and moves leftward.
```

Comment:
its left to right grammar resolution right?

Reply:
`At a high level, a chain-specific address is made of three components separated by colons (`:`) ordered from most specific on the left to most general on the right:`

Reply:
Updated! Resolution does begin on the right.